test/metabase/test/data/users.clj

```diff
@@ -182,15 +177,32 @@
       (when-not user-email
         (throw (ex-info "User does not exist" {:user user})))
       (try
+       (t2/query-one {:update :core_user
+                      :set    {:password      (u.password/hash-bcrypt user-email)
+                               :password_salt ""}
+                      :where  [:= :id user-id]})
+       (apply the-client {:username user-email, :password user-email} args)
+       (finally
         (t2/query-one {:update :core_user
-                       :set    {:password      (u.password/hash-bcrypt user-email)
-                                :password_salt ""}
-                       :where  [:= :id user-id]})
-        (apply client/client {:username user-email, :password user-email} args)
-        (finally
-          (t2/query-one {:update :core_user
-                         :set    old-password-info
-                         :where  [:= :id user-id]}))))))
+                       :set    old-password-info
+                       :where  [:= :id user-id]}))))))
+
+(def user-http-request
+  ^{:doc
+    "A version of our test client that issues the request with credentials for a given User. User may be either a
+    redefined test User name, e.g. `:rasta`, or any User or User ID. (Because we don't have the User's original
+    password, this function temporarily overrides the password for that User.)
```

Comment:
We **do** have the original passwords for all the test users, and overriding passwords seems like it's really going to mess up our ability to parallelize tests using this. There must be a better way of accomplishing this... perhaps create a temp `Session` or something that's non-destructive? Even better, maybe we can just use `with-current-user` directly to bind the relevant things like `*current-user-id*` and then the API request doesn't need to include session headers at all

Reply:
I went with creating a temporary session for the request because it's safe and keeps the mechanism as close as making a real request.